libiop/profiling/boost_profile.cpp

```diff
@@ -0,0 +1,43 @@
+#ifndef CPPDEBUG /* Ubuntu's Boost does not provide binaries compatible with libstdc++'s debug mode so we just reduce functionality here */
+#include <boost/program_options.hpp>
+#endif
+
+#include "libiop/bcs/bcs_common.hpp"
+
+namespace po = boost::program_options;
+using namespace libiop;
+
+typedef struct options{
```

Comment:
Can you put a comment saying that its up to the caller to initialize `hash_enum` from `hash_enum_val`